core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -856,9 +855,12 @@ private void syncExistingInodeMetadata(LockedInodePath inodePath, boolean skipLo
           mPendingPaths.add(child);
           // Update a global counter for all sync streams
           DefaultFileSystemMaster.Metrics.INODE_SYNC_STREAM_PENDING_PATHS_TOTAL.inc();
-          // This asynchronously schedules a job to pre-fetch the statuses into the cache.
-          if (childInode.isDirectory() && mDescendantType == DescendantType.ALL) {
-            mStatusCache.prefetchChildren(child, mMountTable);
+
+          if (mPendingPaths.isQueue()) {
```

Comment:
also I'm thinking if your problem can be solved by just disabling prefetch.

Reply:
PrefetchChild is not the core reason. The core problem is that processing a node will query the ufsStatus of the child node, and put the ufsStatus of the child node in the UfsStatusCache, which can be viewed here. 
```java    
      mStatusCache.prefetchChildren(inodePath.getUri(), mMountTable);
      Collection<UfsStatus> listStatus = mStatusCache
          .fetchChildrenIfAbsent(mRpcContext, inodePath.getUri(), mMountTable);
```
The ufsStatus of these child nodes can only be removed from the cache after the child nodes are synced.
Therefore, it is not enough to simply prohibit prefetchChild, and it is the key to let the child nodes be processed as soon as possible.

In the current implementation, if you want to synchronize a directory with 10,000 subdirectories and 10,000 files in each subdirectory, then, before synchronizing the first file, the ufsStatusCache already has 100 million pieces of data, which takes up a lot of jvm memory( > 70g), and more gc time.

If the stack implementation is used, the number of data in the ufsStatusCache will not exceed 10000 * alluxio.master.metadata.sync.concurrency.level.

Reply:
Yea we should still have prefetch because the UFS operation typically takes longer, therefore we prefer to do the prefetch than fetching on demand.